ir/instr.cpp

```diff
@@ -453,6 +458,30 @@ StateValue BinOp::toSMT(State &s) const {
       return { a.abs(), ap && bp && (b == 0 || a != expr::IntSMin(a.bits())) };
     };
     break;
+
+  case UCmp:
+  case SCmp:
+    fn = [&](auto &a, auto &ap, auto &b, auto &bp) -> StateValue {
+      expr lt, gt;
+      switch (op) {
+      case UCmp:
+        lt = a.ult(b);
+        gt = a.ugt(b);
+        break;
+      case SCmp:
+        lt = a.slt(b);
+        gt = a.sgt(b);
+        break;
+      default:
+        UNREACHABLE();
+      }
+      uint32_t resBits = getType().bits();
+      return {expr::mkIf(std::move(lt), expr::mkInt(-1, resBits),
```

Comment:
one suggestion (sorry, just occurred to me now).
it's more efficient if done like this:
mkIf(a == b, 0, mkIf(lt, -1, 1))

No need for gt and == is way more efficient.